syntax = "proto3";

import "google/protobuf/timestamp.proto";

option csharp_namespace = "TransactionService.Protos";

package transaction;

service TransactionService {
	rpc GetUserTransactions (GetUserTransactionsRequest) returns (TransactionsResponse);
	rpc GetAccountTransactions (GetAccountTransactionsRequest) returns (TransactionsResponse);
	rpc GetTransactionById (GetTransactionByIdRequest) returns (TransactionResponse);
	rpc CreatePaymentTransaction (AddTransactionRequest) returns (TransactionResponse);
	rpc UpdateTransaction (UpdateTransactionRequest) returns (TransactionResponse);
	rpc DeleteTransaction (DeleteTransactionRequest) returns (TransactionResponse);
}

message TransactionReuqest {
	int32 id = 1;
	int32 userId = 2;
	int32 accountId = 3;
	string type = 4;
	float amount = 5;
	// google.protobuf.Timestamp timestamp = 6;
	string status = 7;
}

message NewTransactionRequest {
	int32 userId = 1;
	int32 accountId = 2;
	float amount = 3;
}

message TransactionResponse {
	int32 id = 1;
	int32 userId = 2;
	int32 accountId = 3;
	string type = 4;
	float amount = 5;
	// google.protobuf.Timestamp timestamp = 6;
	string status = 7;
	
	string message = 8;
}

message TransactionsResponse {
	repeated TransactionResponse transactions = 1;

	string message = 2;
}

message Empty {

}

message GetUserTransactionsRequest {
	int32 userId = 1;
}

message GetAccountTransactionsRequest {
	int32 accountId = 1;
}

message GetTransactionByIdRequest {
	int32 id = 1;
}

message AddTransactionRequest {
	NewTransactionRequest transaction = 1;
}

message UpdateTransactionRequest {
	TransactionReuqest transaction = 1;
}

message DeleteTransactionRequest {
	int32 id = 1;
}
